### ISSUE TITLE
Investigate a historical issue:  gzip shouldn't be included by default ?

### ISSUE THREAD

**Issue by laglangyue** [apache/pekko-grpc#286]
I have been interested in FIXME  of GRPC. 
![image](https://github.com/apache/pekko-grpc/assets/35491928/d309304c-eb73-4581-8abf-46c547d010f6)

![image](https://github.com/apache/pekko-grpc/assets/35491928/43f7717c-2fbf-4304-9427-5440a2f07d04)

So I want  to talk about :
（1）Gzip should be the default, or only use Identity?
（2）`supportedCodecs` can be configured by reference?

In my opinion, according to the comments, gzip is not the default value for the client. Users can modify the supported codecs through the conf file

**Comment by laglangyue** [comment]
@raboof  I saw your comments from the PR of history in akka-grpc，WDYT ?

**Comment by Roiocam** [comment]
Seems like it is the same issue https://github.com/grpc/grpc-java/issues/8542?

**Comment by laglangyue** [comment]
> Seems like it is the same issue [grpc/grpc-java#8542](https://github.com/grpc/grpc-java/issues/8542)?

thanks, let me see

**Comment by mdedetrich** [comment]
> Seems like it is the same issue [grpc/grpc-java#8542](https://github.com/grpc/grpc-java/issues/8542)?

If I am reading this correctly this appears to be a non issue creating by a misunderstanding, i.e. `Message-Accept-Encoding` should list all of the compression schemes that you **support**, not the ones that is being used right now.

**Comment by raboof** [comment]
I _think_ the limitation identified here was the fact that we currently do not provide a way for the client to prevent the server from compressing the responses. This could be interesting to allow when making requests from low-powered machines, where you might be happy to spend a bit more bandwidth if it means spending less CPU time decompressing.

No idea why it was added as a 'FIXME' rather than a regular feature request, though - in retrospect is seems rather niche.

**Comment by pjfanning** [comment]
I agree that this is not a major issue. If someone wants to add a config setting that allows users to disable compression support then that is nice to have

**Comment by laglangyue** [comment]
Thanks for all. I only started reading the PR related to grpc-Java.

Yes, aggre that it is a Feature, not FXIME.
`GRPC-Java already support APIs for registering compression methods`
https://github.com/grpc/grpc-java/issues/1704
https://github.com/grpc/grpc-java/pull/9942
